lib/Auth/Process/UpdateAUP.php

```diff
@@ -59,15 +59,23 @@ public function __construct($config, $reserved)
     public function process(&$state)
     {
         if (!empty($this->config['spBlacklist']) && isset($state['SPMetadata']['entityid']) && in_array($state['SPMetadata']['entityid'], $this->config['spBlacklist'], true)) {
-            SimpleSAML_Logger::debug("[aup] process: Skipping blacklisted SP ". var_export($state['SPMetadata']['entityid'], true));
+            SimpleSAML_Logger::debug("[aup:UpdateAUP] Skipping blacklisted SP ". var_export($state['SPMetadata']['entityid'], true));
             return;
         }
         // Check if user is in blacklist
-        if(!empty($this->config['userIdAttribute']) && !empty($this->config['userIdBlacklist']) &&  !empty(array_intersect($state['Attributes'][$this->config['userIdAttribute']], $this->config['userIdBlacklist']))) {
+        if ((!empty($this->config['userIdAttribute']) && !empty($this->config['userIdBlacklist']) && !empty(array_intersect(
+            $state['Attributes'][$this->config['userIdAttribute']],
+            $this->config['userIdBlacklist'])))) {
+            SimpleSAML_Logger::debug("[aup:UpdateAUP] Skipping blacklisted user with id " . $state['Attributes'][$this->config['userIdAttribute']]);
+            return;
+        }
+        // Check if $state['rciamAttributes']['aup'] is empty
+        if (empty($state['rciamAttributes']['aup'])) {
+            SimpleSAML_Logger::debug("[aup:UpdateAUP] No AUP information found in state - skipping");
             return;
         }
         try {
-            SimpleSAML_Logger::debug('[aup] process: ' . $this->config['userIdAttribute'] . ' ' . var_export($state['Attributes'][$this->config['userIdAttribute']],true));
+            SimpleSAML_Logger::debug('[aup:UpdateAUP] ' . $this->config['userIdAttribute'] . ' ' . var_export($state['Attributes'][$this->config['userIdAttribute']],true));
```

Comment:
should we check if the userIdAttribute exists or this has already been checked?

Reply:
I just removed it as it was only for testing purposes actually.